pkg/credentialprovider/plugin/examples/acr/main.go

```diff
@@ -0,0 +1,100 @@
+// +build !providerless
+
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// TODO: this is just a reference implementation, this plugin should be moved
+// to sigs.k8s.io/cloud-provider-azure
+package main
+
+import (
+	"errors"
+	"fmt"
+	"os"
+
+	"k8s.io/klog/v2"
+	"k8s.io/kubernetes/pkg/credentialprovider/apis/criauth/v1alpha1"
+	plugin "k8s.io/kubernetes/pkg/credentialprovider/plugin/framework"
+	"k8s.io/legacy-cloud-providers/azure/auth"
+)
+
+const (
+	azureConfigFileKey = "azure-container-registry-config"
+)
+
+func (a *acrPlugin) GetAuthPluginResponse(image string, extraArgs map[string]string) (*v1alpha1.CRIAuthPluginResponse, error) {
+	configFile, ok := extraArgs[azureConfigFileKey]
+	if !ok {
+		return nil, fmt.Errorf("azure config file argument %s not provided", azureConfigFileKey)
+	}
+
+	f, err := os.Open(configFile)
+	if err != nil {
+		return nil, fmt.Errorf("error reading azure config file: %v", err)
+	}
+	defer f.Close()
+
+	err = a.loadConfig(f)
+	if err != nil {
+		return nil, fmt.Errorf("error loading config: %v", err)
+	}
+
+	a.servicePrincipalToken, err = auth.GetServicePrincipalToken(a.config, a.environment)
+	if err != nil {
+		return nil, fmt.Errorf("error creating service principal token: %v", err)
+	}
+
+	a.registryClient = newAzRegistriesClient(a.config.SubscriptionID, a.environment.ResourceManagerEndpoint, a.servicePrincipalToken)
+
+	if !a.config.UseManagedIdentityExtension {
+		return &v1alpha1.CRIAuthPluginResponse{
+			AuthConfig: v1alpha1.AuthConfig{
+				Username: a.config.AADClientID,
+				Password: a.config.AADClientSecret,
+			},
+		}, nil
+	}
+
+	loginServer := a.parseACRLoginServerFromImage(image)
```

Comment:
is it possible to add back full implementation here? Thanks.
https://github.com/kubernetes/kubernetes/blob/604569482fc34d010a33339f000e75ecbbf345c6/pkg/credentialprovider/azure/azure_credentials.go#L198-L244